torch/cuda/amp/grad_scaler.py

```diff
@@ -20,7 +20,6 @@ class _MultiDeviceReplicator:
     """
 
     def __init__(self, master_tensor: torch.Tensor) -> None:
-        assert master_tensor.is_cuda or master_tensor.device.type == "xla"
```

Comment:
device check is performed inside GradScaler with `_is_tensor_on_supported_device`